src/main/python/ttconv/imsc/attributes.py

```diff
@@ -297,6 +299,26 @@ class TemporalAttributeParsingContext:
   frame_rate: Fraction = Fraction(30, 1)
   tick_rate: int = 1
 
+@dataclass
+class TemporalAttributeWritingContext:
+  frame_rate: Fraction = Fraction(30, 1)
+
+def to_time_format(context: TemporalAttributeWritingContext, time: Fraction) -> str:
+  if context.frame_rate is None:
+    return to_hh_mm_ss_ms(time)
+  return to_frames(context, time)
+
+def to_hh_mm_ss_ms(time: Fraction) -> str:
+  millisecond = (time % 1) * 1000
+  minute, second = divmod(time, 60) 
```

Comment:
@valnoel Don't we have some code somewhere that converts from seconds to HH:MM:SS?

Reply:
Yes, we have such code in the [SRT module](https://github.com/sandflow/ttconv/blob/master/src/main/python/ttconv/srt/time_code.py#L45), and to convert frames to time code as well in the [SCC module](https://github.com/sandflow/ttconv/blob/master/src/main/python/ttconv/scc/time_codes.py#L81).

Isn't it time to merge all this into a common module?

Reply:
What should we do here?
Move the SRT implementation to a generic one and then have the SRT implementation call the generic and the wrap the time into SrtTimeCode

Reply:
In my opinion, if we create a generic time code implementation that handle conversions from/to frames and seconds, we could simply remove the SrtTimeCode and SccTimeCode classes.
But I think we should create a specific issue for this refactoring, don't you think?

Reply:
Based on conversation at standup, we are not going to fix this as part of this PR. 
Logged issue https://github.com/sandflow/ttconv/issues/161